pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -361,6 +415,12 @@ func updateStatusFromPod(taskRun *v1alpha1.TaskRun, pod *corev1.Pod, resourceLis
 		})
 	}
 
+	if len(taskSpecSteps) > 0 {
+		trt := new(TaskRunWithTask)
+		trt.Init(taskRun.Status.Steps, taskSpecSteps[0])
+		sort.Sort(trt)
```

Comment:
Wondering if sorting should be done outside of `updateStatusFromPod` method? :thinking: 
In that way
- Writing the unit test would be easy
- Single responsibility: Sorting responsibility could be taken outside of methods scope

Reply:
I change the code structure a bit in order to separate the sortTaskRun function.